src/BlockArea.cpp

```diff
@@ -441,10 +441,10 @@ bool cBlockArea::Read(cForEachChunkProvider * a_ForEachChunkProvider, int a_MinB
 		LOGWARNING("%s: MaxBlockY less than zero, adjusting to zero", __FUNCTION__);
 		a_MaxBlockY = 0;
 	}
-	else if (a_MaxBlockY > cChunkDef::Height)
+	else if (a_MaxBlockY >= cChunkDef::Height)
 	{
 		LOGWARNING("%s: MaxBlockY more than chunk height, adjusting to chunk height", __FUNCTION__);
-		a_MaxBlockY = cChunkDef::Height;
+		a_MaxBlockY = cChunkDef::Height - 1;
```

Comment:
Thank you, I didn't notice this. I could rename the variables to make it clearer we're working with intervals. Maybe using iterators like naming.
Example: 
    a_MinBlockX becomes a_BeginBlockX
    a_MaxBlockX becomes a_EndBlockX